docs/examples/rails-api/web-external.yaml

```diff
@@ -0,0 +1,124 @@
+---
+#########################################################
+# DEPLOYMENT
+#########################################################
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: web-external
+  labels:
+    application: web-external
+    namespace: some-namespace
+spec:
+  replicas: 2
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+  minReadySeconds: 5
+  selector:
+    matchLabels:
+       application: web-external
+  template:
+    metadata:
+      labels:
+        application: web-external
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - web-external
+            topologyKey: kubernetes.io/hostname
+      containers:
+        - name: web-external
+          # Change this to your image
+          image: some-rails-docker-image
+          imagePullPolicy: Always
+          ports:
+            - name: http
+              containerPort: 3000
+              protocol: TCP
+          envFrom:
+            - configMapRef:
+                name: rails-api-config-map
+          command:
+          - "bundle exec rails s -b 0.0.0.0 -p $PORT"
+          resources:
+            requests:
+              memory: 1Gi
+              cpu: .5
+            limits:
+              memory: 2Gi
+              cpu: 2
+---
+#########################################################
+# HPA
+#########################################################
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: web-external
+  labels:
+    application: web-external
+    namespace: some-namespace
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1beta2
+    kind: Deployment
+    name: web-external
+  minReplicas: 1
+  maxReplicas: 1
```

Comment:
Why HPA with min 1 max 1? it will recreate the pod if it reaches 50% cpu?

Reply:
To reduce the resources required to demo, absolutely could be 1min 2 max